src/main/java/bio/terra/janitor/service/pubsub/TrackedResourceSubscriber.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.janitor.service.pubsub;
+
+import bio.terra.generated.model.CreateResourceRequestBody;
+import bio.terra.janitor.app.configuration.TrackResourcePubsubConfiguration;
+import bio.terra.janitor.common.exception.InvalidMessageException;
+import bio.terra.janitor.service.janitor.JanitorService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Subscribes track resource message and update Janitor record. */
+@Component
+public class TrackedResourceSubscriber {
+  private final TrackResourcePubsubConfiguration trackResourcePubsubConfiguration;
+  private final JanitorService janitorService;
+
+  @Autowired
+  TrackedResourceSubscriber(
+      TrackResourcePubsubConfiguration trackResourcePubsubConfiguration,
+      JanitorService janitorService) {
+    this.trackResourcePubsubConfiguration = trackResourcePubsubConfiguration;
+    this.janitorService = janitorService;
+  }
+
+  public void initialize() {
+    if (!trackResourcePubsubConfiguration.isPubsubEnabled()) {
+      return;
+    }
+
+    createTrackedResourceSubscriber(
+        trackResourcePubsubConfiguration.getProjectId(),
+        trackResourcePubsubConfiguration.getSubscription());
+  }
+
+  private void createTrackedResourceSubscriber(String projectId, String subscriptionId) {
+    ProjectSubscriptionName subscriptionName =
+        ProjectSubscriptionName.of(projectId, subscriptionId);
+
+    // Instantiate an asynchronous message receiver.
+    MessageReceiver receiver =
+        (PubsubMessage message, AckReplyConsumer consumer) -> {
+          // Handle incoming message, then always ack the received message.
+          try {
+            CreateResourceRequestBody body =
+                new ObjectMapper()
+                    .readValue(message.getData().toStringUtf8(), CreateResourceRequestBody.class);
+            janitorService.createResource(body);
+          } catch (JsonProcessingException e) {
+            throw new InvalidMessageException(
+                "Invalid track resource pubsub message: " + message.toString(), e);
+          } finally {
+            consumer.ack();
```

Comment:
I don't think we should always ack the message. We should only ack it if we've persisted correctly. If we throw a database exception, the message should not be acked.

I think we should set up a dead letter queue for messages we're not able to process of silently dropping them. Then we can add monitoring/alerting and see what ends up there. https://cloud.google.com/pubsub/docs/dead-letter-topics
Even invalid format message should not be dropped silently - what if there's a bug in the janitor's processing of the request?
Leave a TODO for now instead of implementing the DLQ to not make this PR too big, but only ack if we're successful.